New warnings on 20230322

x32, gcc 12.2.0-14, glibc 2.36-8

```
cc -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -ffile-prefix-map=/home/nabijaczleweli/code/build-output/neomutt-20230322+dfsg=. -specs=/usr/share/dpkg/pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include/p11-kit-1 -I/usr/include -isystem /usr/include/mit-krb5 -I. -I.. -Wall -g
-O2 -ffile-prefix-map=/home/nabijaczleweli/code/build-output/neomutt-20230322+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -I../test -MT imap/message.o -MD -MP -MF imap/message.Tpo -c -o imap/message.o ../imap/message.c
../imap/message.c: In function ‘imap_msg_open’:
../imap/message.c:2104:3: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2104 |   fgets(buf, sizeof(buf), msg->fp);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../imap/message.c:2108:5: warning: ignoring return value of ‘fgets’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2108 |     fgets(buf, sizeof(buf), msg->fp);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
from a rudimentary analysis (i typed /buf in vi and then n a few times) it appears that buf isn't really used here, and this is a line-counting lseek(SEEK_END); `(void)` or some other mechanism to mark this used is probably sufficient.

```
cc -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -ffile-prefix-map=/home/nabijaczleweli/code/build-output/neomutt-20230322+dfsg=. -specs=/usr/share/dpkg/pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include/p11-kit-1 -I/usr/include -isystem /usr/include/mit-krb5 -I. -I.. -Wall -g
-O2 -ffile-prefix-map=/home/nabijaczleweli/code/build-output/neomutt-20230322+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -I../test -MT mutt/slist.o -MD -MP -MF mutt/slist.Tpo -c -o mutt/slist.o ../mutt/slist.c
../mutt/regex.c: In function ‘mutt_replacelist_apply’:
../mutt/regex.c:421:15: warning: ‘__builtin_strncpy’ output may be truncated copying between 0 and 1023 bytes from a string of length 2047 [-Wstringop-truncation]
  421 |               strncpy(&dst[tlen], src, cpysize);
      |               ^
```
No clue if this means anything; it appears to be a false positive because, unlike the strncpy on L428, this doesn't subslice `src`.

--- a/imap/message.c
+++ b/imap/message.c
@@ -2106,11 +2106,9 @@ parsemsg:
   }
 
   e->lines = 0;
-  fgets(buf, sizeof(buf), msg->fp);
-  while (!feof(msg->fp))
+  while (fgets(buf, sizeof(buf), msg->fp) && !feof(msg->fp))
   {
     e->lines++;
-    fgets(buf, sizeof(buf), msg->fp);
   }
 
   e->body->length = ftell(msg->fp) - e->body->offset;

